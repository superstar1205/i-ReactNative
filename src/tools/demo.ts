export enum CommentType {
  REMOVE_FILE = `@demo remove-file`,
  REMOVE_CURRENT_LINE = `@demo remove-current-line`,
  REMOVE_NEXT_LINE = `@demo remove-next-line`,
  REMOVE_BLOCK_START = `@demo remove-block-start`,
  REMOVE_BLOCK_END = `@demo remove-block-end`,
}

/**
 * Take the file content as a string and remove any
 * line of code with an `// @demo remove-current-line` comment
 */
function removeCurrentLine(contents: string, comment = CommentType.REMOVE_CURRENT_LINE): string {
  const lines = contents.split("\n")
  const result = lines.filter((line) => !line.includes(comment))
  return result.join("\n")
}

/**
 * Take the file content as a string and remove the next line
 * of code with an `// @demo remove-next-line` comment before it
 */
function removeNextLine(contents: string, comment = CommentType.REMOVE_NEXT_LINE): string {
  const lines = contents.split("\n")
  const result = lines.filter((line, index) => {
    const prevLine = lines[index - 1]

    const preserveCurrent = line.includes(comment) === false
    const preservePrevious = prevLine !== undefined && prevLine.includes(comment) === false

    if (index === 0) {
      // if we are on the first line, there is no previous line to check
      return preserveCurrent
    }

    // keep current line if there is no comment in current or previous line
    const keepLine = preserveCurrent && preservePrevious
    return keepLine
  })
  return result.join("\n")
}

/**
 * Take the file content as a string and remove the lines of code between
 * `// @demo remove-block-start` and `// @demo remove-block-end` comments
 */
function removeBlock(
  contents: string,
  comment = { start: CommentType.REMOVE_BLOCK_START, end: CommentType.REMOVE_BLOCK_END },
): string {
  const { start, end } = comment
  const lines = contents.split("\n")

  const findIndex = (l: typeof lines, c: typeof start | typeof end) =>
    l.findIndex((line) => line.includes(c))
  const NOT_FOUND = -1

  const blockStartIndex = findIndex(lines, start)
  const blockEndIndex = findIndex(lines, end)
  const blockExists = findIndex(lines, start) !== NOT_FOUND && blockEndIndex !== NOT_FOUND

  if (blockExists) {
    const blockLength = blockEndIndex - blockStartIndex + 1
    lines.splice(blockStartIndex, blockLength) // mutates `lines`
  }

  const updateContents = lines.join("\n")

  const anotherBlockExists =
    findIndex(lines, start) !== NOT_FOUND && findIndex(lines, end) !== NOT_FOUND
  if (anotherBlockExists) {
    return removeBlock(updateContents, comment)
  }

  return updateContents
}

/**
 * Perform all remove operations possible in a file
 * @param contents The file contents as a string
 * @return The file contents with all remove operations performed
 */
function remove(contents: string): string {
  const result = removeBlock(removeNextLine(removeCurrentLine(contents)))
  return result
}

export const demo = {
  CommentType,
  removeCurrentLine,
  removeNextLine,
  removeBlock,
  remove,
} as const
