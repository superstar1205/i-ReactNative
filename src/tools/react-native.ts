import { GluegunToolbox } from "gluegun"
import { children } from "./filesystem-ext"

export const isAndroidInstalled = (toolbox: GluegunToolbox): boolean => {
  const androidHome = process.env.ANDROID_HOME
  const hasAndroidEnv = !toolbox.strings.isBlank(androidHome)
  const hasAndroid = hasAndroidEnv && toolbox.filesystem.exists(`${androidHome}/tools`) === "dir"

  return Boolean(hasAndroid)
}

type CopyBoilerplateOptions = {
  boilerplatePath: string
  projectName: string
  excluded: Array<string>
}

/**
 * Copies the boilerplate over to the destination folder.
 *
 */
export async function copyBoilerplate(toolbox: GluegunToolbox, options: CopyBoilerplateOptions) {
  const { filesystem } = toolbox
  const { copyAsync, path } = filesystem

  // ensure the destination folder exists
  await filesystem.dirAsync(options.projectName)

  // rather than copying everything wholesale, let's check what's in the boilerplate folder
  // and copy over everything except stuff like lockfiles and node_modules
  // just to make it faster, y'know? Don't want to copy unnecessary stuff
  const filesAndFolders = children(options.boilerplatePath, true)
  const copyTargets = filesAndFolders.filter(
    (file) => !options.excluded.find((exclusion) => file.includes(exclusion)),
  )

  // kick off a bunch of copies
  const copyPromises = copyTargets.map((fileOrFolder) =>
    copyAsync(path(options.boilerplatePath, fileOrFolder), path(options.projectName, fileOrFolder)),
  )

  // copy them all at once
  return Promise.all(copyPromises)
}

export async function renameReactNativeApp(
  toolbox: GluegunToolbox,
  oldName: string,
  newName: string,
  oldBundleIdentifier: string,
  newBundleIdentifier: string,
) {
  const { parameters, filesystem, print, strings } = toolbox
  const { kebabCase } = strings
  const { path } = filesystem

  // debug?
  const debug = Boolean(parameters.options.debug)
  const log = <T = unknown>(m: T): T => {
    debug && print.info(` ${m}`)
    return m
  }

  // lower case stuff
  const oldnamelower = oldName.toLowerCase()
  const newnamelower = newName.toLowerCase()

  // kebab case
  const oldnamekebab = kebabCase(oldName)
  const newnamekebab = kebabCase(newName)

  // SCREAMING_SNAKE_CASE
  const oldnamesnake = oldnamelower.replace(/[^a-z0-9]/g, "_").toUpperCase()
  const newnamesnake = newnamelower.replace(/[^a-z0-9]/g, "_").toUpperCase()

  async function rename(oldFile: string, newFile: string) {
    log(`Renaming ${oldFile} to ${newFile}`)
    return filesystem.renameAsync(oldFile, newFile)
  }

  // rename files and folders

  // prettier-ignore
  await Promise.allSettled([
    rename(`android/app/src/main/java/com/${oldnamelower}`, newnamelower),
    rename(`android/app/src/debug/java/com/${oldnamelower}`, newnamelower),
    rename(`ios/${oldName}.xcodeproj/xcshareddata/xcschemes/${oldName}.xcscheme`, `${newName}.xcscheme`),
    rename(`ios/${oldName}Tests/${oldName}Tests.m`, `${newName}Tests.m`),
    rename(`ios/${oldName}-Bridging-Header.h`, `${newName}-Bridging-Header.h`),
    rename(`ios/${oldName}.xcworkspace`, `${newName}.xcworkspace`),
    rename(`ios/${oldName}`, `${newName}`),
  ])

  // these we delay to avoid race conditions
  await Promise.allSettled([
    rename(`ios/${oldName}Tests`, `${newName}Tests`),
    rename(`ios/${oldName}.xcodeproj`, `${newName}.xcodeproj`),
  ])

  // if the bundle identifier / android package name changed,
  // we need to move everything to the new folder structure
  const oldPath = oldBundleIdentifier.replace(/\./g, "/")
  const newPath = newBundleIdentifier.replace(/\./g, "/")

  if (oldBundleIdentifier !== newBundleIdentifier) {
    log(`Renaming bundle identifier to ${newBundleIdentifier}`)

    // move everything at the old bundle identifier path to the new one
    await Promise.allSettled([
      moveDeepFolder(
        toolbox,
        `android/app/src/main/java/${oldPath}`,
        `android/app/src/main/java/${newPath}`,
      ),
      moveDeepFolder(
        toolbox,
        `android/app/src/debug/java/${oldPath}`,
        `android/app/src/debug/java/${newPath}`,
      ),
    ])
  }

  // here's a list of all the files to patch the name in
  const filesToPatch = [
    `app.json`,
    `package.json`,
    `index.js`,
    `android/settings.gradle`,
    `android/app/_BUCK`,
    `android/app/BUCK`,
    `android/app/build.gradle`,
    `android/app/src/debug/java/${newPath}/ReactNativeFlipper.java`,
    `android/app/src/main/AndroidManifest.xml`,
    `android/app/src/main/java/${newPath}/MainActivity.java`,
    `android/app/src/main/java/${newPath}/MainApplication.java`,
    `android/app/src/main/java/${newPath}/MainApplication.java`,
    `android/app/src/main/java/${newPath}/newarchitecture/MainApplicationReactNativeHost.java`,
    `android/app/src/main/java/${newPath}/newarchitecture/components/MainComponentsRegistry.java`,
    `android/app/src/main/java/${newPath}/newarchitecture/modules/MainApplicationTurboModuleManagerDelegate.java`,
    `android/app/src/main/jni/Android.mk`,
    `android/app/src/main/jni/MainApplicationTurboModuleManagerDelegate.h`,
    `android/app/src/main/jni/MainComponentsRegistry.h`,
    `android/app/src/main/res/values/strings.xml`,
    `ios/Podfile`,
    `ios/${newName}/Info.plist`,
    `ios/${newName}.xcodeproj/project.pbxproj`,
    `ios/${newName}.xcodeproj/xcshareddata/xcschemes/${newName}.xcscheme`,
    `ios/${newName}.xcworkspace/contents.xcworkspacedata`,
    `ios/${newName}Tests/${newName}Tests.m`,
    `ios/${newName}/AppDelegate.mm`,
    `ios/${newName}/LaunchScreen.storyboard`,
  ]

  // patch the files
  await Promise.allSettled(
    filesToPatch.map(async (file) => {
      // no need to patch files that don't exist
      const exists = await filesystem.existsAsync(path(file))
      if (!exists) return

      const content = await filesystem.readAsync(path(process.cwd(), file), "utf8")

      log(`Patching ${file} - ${oldName} to ${newName} and variants`)

      // replace all instances of the old name and all its variants
      const newContent = content
        .replace(new RegExp(oldBundleIdentifier, "g"), newBundleIdentifier)
        .replace(new RegExp(oldnamekebab, "g"), newnamekebab)
        .replace(new RegExp(oldnamesnake, "g"), newnamesnake)
        .replace(new RegExp(oldName, "g"), newName)
        .replace(new RegExp(oldnamelower, "g"), newnamelower)

      // write the new content back to the file
      await filesystem.writeAsync(file, newContent, { atomic: true })
    }),
  )
}

async function moveDeepFolder(toolbox: GluegunToolbox, oldPath: string, newPath: string) {
  const { filesystem } = toolbox
  const { path } = filesystem

  // ensure newPath exists
  await filesystem.dirAsync(newPath)

  // move the files
  const filesToMove = children(oldPath)
  const movePromises = filesToMove.map((file) =>
    filesystem.moveAsync(path(oldPath, file), path(newPath, file)),
  )

  await Promise.allSettled(movePromises)

  // remove the old folder
  await filesystem.removeAsync(oldPath)
}
