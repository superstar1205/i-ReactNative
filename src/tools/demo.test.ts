import { demo } from "./demo"

describe("demo", () => {
  describe("removeCurrentLine", () => {
    it(`should remove line with "// ${demo.CommentType.REMOVE_CURRENT_LINE}" comment`, () => {
      const contents = `
        import React from "react"
        import { StyleProp, View, ViewStyle } from "react-native"
        
        interface DemoDividerProps {
            type?: "vertical" | "horizontal"
            size?: number
            style?: StyleProp<ViewStyle>
        }
        
        export function DemoDivider(props: DemoDividerProps) {
            const { type = "horizontal", size = 10, style: $styleOverride } = props
        
            return (
              <View
                  style={[
                    $divider,
                    type === "horizontal" && { height: size },
                    type === "vertical" && { width: size },
                    $styleOverride,
                  ]}
              />
            )
        }
        
        const $divider: ViewStyle = {
            flexGrow: 0,
            flexShrink: 0, // ${demo.CommentType.REMOVE_CURRENT_LINE}
        }
      `
      const result = demo.removeCurrentLine(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_CURRENT_LINE)
    })

    it(`should remove line with "/* ${demo.CommentType.REMOVE_CURRENT_LINE} */" comment`, () => {
      const contents = `
        import React from "react"
        import { StyleProp, View, ViewStyle } from "react-native"
        
        interface DemoDividerProps {
            type?: "vertical" | "horizontal"
            size?: number
            style?: StyleProp<ViewStyle>
        }
        
        export function DemoDivider(props: DemoDividerProps) {
            const { type = "horizontal", size = 10, style: $styleOverride } = props
        
            return (
              <View
                  style={[
                    $divider,
                    type === "horizontal" && { height: size },
                    type === "vertical" && { width: size },
                    $styleOverride, {/* ${demo.CommentType.REMOVE_CURRENT_LINE} */}
                  ]}
              />
            )
        }
        
        const $divider: ViewStyle = {
            flexGrow: 0,
            flexShrink: 0,
        }
      `
      const result = demo.removeCurrentLine(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_CURRENT_LINE)
    })
  })

  describe("removeNextLine", () => {
    it(`should remove comment and next line after "// ${demo.CommentType.REMOVE_NEXT_LINE}"`, () => {
      const contents = `
        export * from "./WelcomeScreen"
        export * from "./LoginScreen"
        // ${demo.CommentType.REMOVE_NEXT_LINE}
        export * from "./DemoCommunityScreen"
      `
      const result = demo.removeNextLine(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_NEXT_LINE)
      expect(result).not.toContain("DemoCommunityScreen")
    })

    it(`should remove comment and next line after "/* ${demo.CommentType.REMOVE_NEXT_LINE} */"`, () => {
      const contents = `
        export * from "./WelcomeScreen"
        export * from "./LoginScreen"
        /* ${demo.CommentType.REMOVE_NEXT_LINE} */
        export * from "./DemoCommunityScreen"
      `
      const result = demo.removeNextLine(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_NEXT_LINE)
      expect(result).not.toContain("DemoCommunityScreen")
    })
  })

  describe("removeBlock", () => {
    it(`should remove comments and lines between "// ${demo.CommentType.REMOVE_BLOCK_START}" and "// ${demo.CommentType.REMOVE_BLOCK_END}"`, () => {
      const contents = `
        export * from "./WelcomeScreen"
        export * from "./LoginScreen"
        // ${demo.CommentType.REMOVE_BLOCK_START}
        export * from "./DemoCommunityScreen"
        export * from "./DemoDebugScreen"
        export * from "./DemoComponentsScreen/DemoComponentsScreen"
        // ${demo.CommentType.REMOVE_BLOCK_END}
        export * from "./ErrorScreen/ErrorBoundary"
        // export other screens here
      `
      const result = demo.removeBlock(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_BLOCK_START)
      expect(result).not.toContain(demo.CommentType.REMOVE_BLOCK_END)
      expect(result).not.toContain("DemoCommunityScreen")
      expect(result).not.toContain("DemoDebugScreen")
      expect(result).not.toContain("DemoComponentsScreen")
    })

    it(`should remove comments and lines between "/* ${demo.CommentType.REMOVE_BLOCK_START} */" and "/* ${demo.CommentType.REMOVE_BLOCK_END} */"`, () => {
      const contents = `
        export * from "./WelcomeScreen"
        export * from "./LoginScreen"
        /* ${demo.CommentType.REMOVE_BLOCK_START} */
        export * from "./DemoCommunityScreen"
        export * from "./DemoDebugScreen"
        export * from "./DemoComponentsScreen/DemoComponentsScreen"
        /* ${demo.CommentType.REMOVE_BLOCK_END} */
        export * from "./ErrorScreen/ErrorBoundary"
        // export other screens here
      `
      const result = demo.removeBlock(contents)
      expect(result).toMatchSnapshot()
      expect(result).not.toContain(demo.CommentType.REMOVE_BLOCK_START)
      expect(result).not.toContain(demo.CommentType.REMOVE_BLOCK_END)
      expect(result).not.toContain("DemoCommunityScreen")
      expect(result).not.toContain("DemoDebugScreen")
      expect(result).not.toContain("DemoComponentsScreen")
    })
  })
})
