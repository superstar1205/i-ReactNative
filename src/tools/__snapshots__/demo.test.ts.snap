// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo remove should remove all comments in AppNavigator 1`] = `
"
/**
 * The app navigator (formerly \\"AppNavigator\\" and \\"MainNavigator\\") is used for the primary
 * navigation flows of your app.
 * Generally speaking, it will contain an auth flow (registration, login, forgot password)
 * and a \\"main\\" flow which the user will use once logged in.
 */
 import {
  DarkTheme,
  DefaultTheme,
  NavigationContainer,
} from \\"@react-navigation/native\\"
import { createNativeStackNavigator } from \\"@react-navigation/native-stack\\"
import { StackScreenProps } from \\"@react-navigation/stack\\"
import { observer } from \\"mobx-react-lite\\"
import React from \\"react\\"
import { useColorScheme } from \\"react-native\\"
import Config from \\"../config\\"
import { 
  WelcomeScreen 
} from \\"../screens\\"
import { navigationRef, useBackButtonHandler } from \\"./navigation-utilities\\"

/**
 * This type allows TypeScript to know what routes are defined in this navigator
 * as well as what properties (if any) they might take when navigating to them.
 *
 * If no params are allowed, pass through \`undefined\`. Generally speaking, we
 * recommend using your MobX-State-Tree store(s) to keep application state
 * rather than passing state through navigation params.
 *
 * For more information, see this documentation:
 *   https://reactnavigation.org/docs/params/
 *   https://reactnavigation.org/docs/typescript#type-checking-the-navigator
 *   https://reactnavigation.org/docs/typescript/#organizing-types
 */
export type AppStackParamList = {
  Welcome: undefined
  // ðŸ”¥ Your screens go here
}

/**
 * This is a list of all the route names that will exit the app if the back button
 * is pressed while in that screen. Only affects Android.
 */
const exitRoutes = Config.exitRoutes

export type AppStackScreenProps<T extends keyof AppStackParamList> = StackScreenProps<
  AppStackParamList,
  T
>

// Documentation: https://reactnavigation.org/docs/stack-navigator/
const Stack = createNativeStackNavigator<AppStackParamList>()

const AppStack = observer(function AppStack() {
  return (
    <Stack.Navigator
      screenOptions={{ headerShown: false }}
    >
          <Stack.Screen name=\\"Welcome\\" component={WelcomeScreen} />
      {/** ðŸ”¥ Your screens go here */}
    </Stack.Navigator>
  )
})

interface NavigationProps extends Partial<React.ComponentProps<typeof NavigationContainer>> {}

export const AppNavigator = observer(function AppNavigator(props: NavigationProps) {
  const colorScheme = useColorScheme()

  useBackButtonHandler((routeName) => exitRoutes.includes(routeName))

  return (
    <NavigationContainer
      ref={navigationRef}
      theme={colorScheme === \\"dark\\" ? DarkTheme : DefaultTheme}
      {...props}
    >
      <AppStack />
    </NavigationContainer>
  )
})
"
`;

exports[`demo remove should remove all comments in WelcomeScreen 1`] = `
"
import { observer } from \\"mobx-react-lite\\"
import React, { useLayoutEffect } from \\"react\\"
import { Image, ImageStyle, TextStyle, View, ViewStyle } from \\"react-native\\"
import { SafeAreaView } from \\"react-native-safe-area-context\\"
import { Button, Header, Text } from \\"../components\\"
import { isRTL } from \\"../i18n\\"
import { colors, spacing } from \\"../theme\\"

const welcomeLogo = require(\\"../../assets/images/logo.png\\")
const welcomeFace = require(\\"../../assets/images/welcome-face.png\\")


export const WelcomeScreen = observer(function WelcomeScreen(
  ) {

  return (
    <View style={$container}>
      <View style={$topContainer}>
        <Image style={$welcomeLogo} source={welcomeLogo} resizeMode=\\"contain\\" />
        <Text style={$welcomeHeading} tx=\\"welcomeScreen.readyForLaunch\\" preset=\\"heading\\" />
        <Text tx=\\"welcomeScreen.exciting\\" preset=\\"subheading\\" />
        <Image style={$welcomeFace} source={welcomeFace} resizeMode=\\"contain\\" />
      </View>

      <SafeAreaView style={$bottomContainer} edges={[\\"bottom\\"]}>
        <View style={$bottomContentContainer}>
          <Text tx=\\"welcomeScreen.postscript\\" size=\\"md\\" />
        </View>
      </SafeAreaView>
    </View>
  )
})

const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.background,
}

const $topContainer: ViewStyle = {
  flexShrink: 1,
  flexGrow: 1,
  flexBasis: \\"57%\\",
  justifyContent: \\"center\\",
  paddingHorizontal: spacing.large,
}

const $bottomContainer: ViewStyle = {
  flexShrink: 1,
  flexGrow: 0,
  flexBasis: \\"43%\\",
  backgroundColor: colors.palette.neutral100,
  borderTopLeftRadius: 16,
  borderTopRightRadius: 16,
}

const $bottomContentContainer: ViewStyle = {
  flex: 1,
  paddingHorizontal: spacing.large,
  justifyContent: \\"space-around\\",
}

const $welcomeLogo: ImageStyle = {
  height: 88,
  width: \\"100%\\",
  marginBottom: spacing.huge,
}

const $welcomeFace: ImageStyle = {
  height: 169,
  width: 269,
  position: \\"absolute\\",
  bottom: -47,
  right: -80,
  transform: [{ scaleX: isRTL ? -1 : 1 }],
}

const $welcomeHeading: TextStyle = {
  marginBottom: spacing.medium,
}
"
`;

exports[`demo removeBlock should remove comments and lines between "/* @demo remove-block-start */" and "/* @demo remove-block-end */" 1`] = `
"
        export * from \\"./WelcomeScreen\\"
        export * from \\"./LoginScreen\\"
        export * from \\"./ErrorScreen/ErrorBoundary\\"
        // export other screens here
      "
`;

exports[`demo removeBlock should remove comments and lines between "// @demo remove-block-start" and "// @demo remove-block-end" 1`] = `
"
        export * from \\"./WelcomeScreen\\"
        export * from \\"./LoginScreen\\"
        export * from \\"./ErrorScreen/ErrorBoundary\\"
        // export other screens here
      "
`;

exports[`demo removeCurrentLine should remove line with "/* @demo remove-current-line */" comment 1`] = `
"
        import React from \\"react\\"
        import { StyleProp, View, ViewStyle } from \\"react-native\\"
        
        interface DemoDividerProps {
            type?: \\"vertical\\" | \\"horizontal\\"
            size?: number
            style?: StyleProp<ViewStyle>
        }
        
        export function DemoDivider(props: DemoDividerProps) {
            const { type = \\"horizontal\\", size = 10, style: $styleOverride } = props
        
            return (
              <View
                  style={[
                    $divider,
                    type === \\"horizontal\\" && { height: size },
                    type === \\"vertical\\" && { width: size },
                  ]}
              />
            )
        }
        
        const $divider: ViewStyle = {
            flexGrow: 0,
            flexShrink: 0,
        }
      "
`;

exports[`demo removeCurrentLine should remove line with "// @demo remove-current-line" comment 1`] = `
"
        import React from \\"react\\"
        import { StyleProp, View, ViewStyle } from \\"react-native\\"
        
        interface DemoDividerProps {
            type?: \\"vertical\\" | \\"horizontal\\"
            size?: number
            style?: StyleProp<ViewStyle>
        }
        
        export function DemoDivider(props: DemoDividerProps) {
            const { type = \\"horizontal\\", size = 10, style: $styleOverride } = props
        
            return (
              <View
                  style={[
                    $divider,
                    type === \\"horizontal\\" && { height: size },
                    type === \\"vertical\\" && { width: size },
                    $styleOverride,
                  ]}
              />
            )
        }
        
        const $divider: ViewStyle = {
            flexGrow: 0,
        }
      "
`;

exports[`demo removeNextLine should remove comment and next line after "/* @demo remove-next-line */" 1`] = `
"
        export * from \\"./WelcomeScreen\\"
        export * from \\"./LoginScreen\\"
      "
`;

exports[`demo removeNextLine should remove comment and next line after "// @demo remove-next-line" 1`] = `
"
        export * from \\"./WelcomeScreen\\"
        export * from \\"./LoginScreen\\"
      "
`;
